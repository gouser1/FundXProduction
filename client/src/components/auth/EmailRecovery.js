import React from 'react';
import { withRouter } from 'react-router-dom';
import {
  Grid,
  Box,
  Button,
  Paper,
  withStyles,
  IconButton,
  TextField,
  InputAdornment,
  Typography,
} from '@material-ui/core';
import LogoNav from '../../images/home/LogoNav.png';
import Email from '@material-ui/icons/Email';
import useStyles from './authStyle';

const EmailConfirm = (props) => {
  const { history } = props;

  const handleLinkClick = (pageURL) => {
    history.push(pageURL);
  };

  const classes = useStyles();

  const LogoButton = withStyles(() => ({}))(IconButton);

  return (
    <div className={classes.root}>
      <Grid
        container
        spacing={0}
        direction='column'
        alignItems='center'
        justify='center'
        style={{ minHeight: '100vh' }}
      >
        <LogoButton
          onClick={() => handleLinkClick('/')}
          style={{ backgroundColor: 'transparent' }}
        >
          <img src={LogoNav} alt='' width='130px' height='35px' />
        </LogoButton>
        <Paper elevation={10} className={classes.paper4}>
          <h1 className={classes.h1}>Recover Password</h1>
          <Box pb={1}>
            <h2 className={classes.h2}>
              Please enter your email address to request a password reset.
            </h2>{' '}
          </Box>
          <Box pt={5} pb={3}>
            {' '}
            <TextField
              id='email'
              label={
                <Typography className={classes.textfield}>Email</Typography>
              }
              variant='outlined'
              color='primary'
              fullWidth
              InputProps={{
                endAdornment: (
                  <InputAdornment position='end'>
                    <Email style={{ fill: '#c8c8c8' }} />
                  </InputAdornment>
                ),
              }}
            />
          </Box>

          <Grid item xs={12}>
            <Box textAlign='center'>
              <Button
                fullWidth
                onClick={() => handleLinkClick('/')}
                className={classes.button}
              >
                Reset Password
              </Button>
            </Box>
          </Grid>
        </Paper>
      </Grid>
    </div>
  );
};

export default withRouter(EmailConfirm);
