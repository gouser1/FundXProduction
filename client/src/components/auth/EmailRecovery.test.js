import React from 'react';
import renderer from 'react-test-renderer';

import EmailRecovery from './EmailRecovery';

it('renders correctly', () => {
  const tree = renderer.create(<EmailRecovery />).toJSON();
  expect(tree).toMatchSnapshot();
});
