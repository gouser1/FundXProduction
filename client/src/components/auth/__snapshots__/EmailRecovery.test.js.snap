// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          height="35px"
          src="LogoNav.png"
          width="130px"
        />
      </span>
    </button>
    <div
      className="MuiPaper-root makeStyles-paper4-10 MuiPaper-elevation10 MuiPaper-rounded"
    >
      <h1
        className="makeStyles-h1-5"
      >
        Recover Password
      </h1>
      <div
        className="MuiBox-root MuiBox-root-13"
      >
        <h2
          className="makeStyles-h2-6"
        >
          Please enter your email address to request a password reset.
        </h2>
         
      </div>
      <div
        className="MuiBox-root MuiBox-root-14"
      >
         
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            <p
              className="MuiTypography-root makeStyles-textfield-12 MuiTypography-body1"
            >
              Email
            </p>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              disabled={false}
              id="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "fill": "#c8c8c8",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-17"
              >
                <span>
                  <p
                    className="MuiTypography-root makeStyles-textfield-12 MuiTypography-body1"
                  >
                    Email
                  </p>
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiBox-root MuiBox-root-19"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Reset Password
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
